external/llvm-project/mlir/lib/Dialect/Async/IR/Async.cpp

```diff
@@ -55,11 +55,177 @@ YieldOp::getMutableSuccessorOperands(Optional<unsigned> index) {
 }
 
 //===----------------------------------------------------------------------===//
-/// ExecuteOp
+/// LaunchOp
 //===----------------------------------------------------------------------===//
 
+constexpr char kCallee[] = "callee";
 constexpr char kOperandSegmentSizesAttr[] = "operand_segment_sizes";
 
+void LaunchOp::build(OpBuilder &builder, OperationState &result, FuncOp func,
+                     ValueRange dependencies, ValueRange operands) {
+  // set callee
+  result.addAttribute(kCallee, SymbolRefAttr::get(func));
```

Comment:
Instead of making this a string constant, there's `calleeAttributeName(result.name)` - and there'll be analogous methods for operand sized attr segments